### DISCUSSION TITLE
Support nested version control system files

### DISCUSSION THREAD

**Discussion by DanielNoord** [biomejs/biome#3530] [Ideas]
This was originally tried in https://github.com/biomejs/biome/pull/1134 but later reverted in https://github.com/biomejs/biome/pull/1249. See also discussion inhttps://github.com/biomejs/biome/issues/805

Basically, it would be good if `.gitignore` or other VCS ignore files that exist in subdirectories would also be respected.

**Comment by dyc3** [comment]
This is definitely related to #2228. This is also likely a problem for `.editorconfig` files too.

**Comment by gscalise** [comment]
Is there any reason in the TODO in here: https://github.com/biomejs/biome/blob/b260d5b0e6b2e8e0093128ca2bf1f66c8e5d35da/crates/biome_fs/src/fs/os.rs#L219-L222 to propose "directly using .gitignore" instead of relying on `git2-rs` and letting the library figure out what to ignore? 

As far as I understand, there's a multitude of ways to ignore files in a git repository (`.gitignore`, nested `.gitignore`, entries in `.git/info/exclude`, global settings, user settings, etc). Trying to replicate `git`'s behavior by only handling a subset of these cases -when it can be delegated to the library- doesn't seem too promising IMO.
